apps/antalmanac/src/components/RightPane/CoursePane/CoursePaneButtonRow.tsx

```diff
@@ -58,24 +60,14 @@ function renderEmptySelectValue() {
  * e.g. show/hide the section code, instructors, etc.
  */
 export function ColumnToggleButton() {
-    const [activeColumns, setActiveColumns] = useState(RightPaneStore.getActiveColumns());
+    const { selectedColumns, setSelectedColumns } = useColumnStore();
```

Comment:
You can also slice the store to reduce the amount of data the component subscribes to by default. The `useStore` function accepts a callback which can return specific properties from the store. You can return the setter/getter for columns in an array, sort of like a regular React `useState`.

```suggestion
    const [ selectedColumns, setSelectedColumns ] = useColumnStore(store => [store.selectedColumns, store.setSelectedColumns]);
```

Reply:
Interesting. What's the benefit of that?

Reply:
If you don't slice, then the component will react to all changes within the store (which isn't necessarily a big deal if you manage small stores properly, but generally an anti-pattern). [Reference](https://docs.pmnd.rs/zustand/recipes/recipes#fetching-everything)